util/webhook/webhook.go

```diff
@@ -212,44 +212,45 @@ func (a *ArgoCDWebhookHandler) HandleEvent(payload interface{}) {
 	for _, webURL := range webURLs {
 		log.Infof("Received push event repo: %s, revision: %s, touchedHead: %v", webURL, revision, touchedHead)
 	}
-	appIf := a.appClientset.ArgoprojV1alpha1().Applications(a.ns)
-	apps, err := appIf.List(context.Background(), metav1.ListOptions{})
-	if err != nil {
-		log.Warnf("Failed to list applications: %v", err)
-		return
-	}
-
-	trackingMethod, err := a.settingsSrc.GetTrackingMethod()
-	if err != nil {
-		log.Warnf("Failed to get trackingMethod: %v", err)
-		return
-	}
-	appInstanceLabelKey, err := a.settingsSrc.GetAppInstanceLabelKey()
-	if err != nil {
-		log.Warnf("Failed to get appInstanceLabelKey: %v", err)
-		return
-	}
 
-	for _, webURL := range webURLs {
-		repoRegexp, err := getWebUrlRegex(webURL)
+	for _, namespace := range(a.namespaces) {
```

Comment:
`a.ArgoCDServerOpts.ApplicationNamespaces` can include glob patterns. So we need to list _all_ apps, and then filter.